ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -865,98 +870,82 @@ def set_waypoints_velocity(self):
                 min(self.default_velocity, self.waypoints[self.final_waypoints_start_ptr].get_maxV()):
                 # handle case where car is at target speed and no
                 # traffic lights within stopping distance
-            if self.state != 'maintainspeed':
-                rospy.logwarn("Set car state to maintainspeed at ptr = {}".format(self.final_waypoints_start_ptr))
-                self.state = 'maintainspeed'
-            self.maintain_speed(self.final_waypoints_start_ptr, self.
+                if self.state != 'maintainspeed':
+                    rospy.logwarn("Set car state to maintainspeed at ptr = {}".format(self.final_waypoints_start_ptr))
+                    self.state = 'maintainspeed'
+                self.maintain_speed(self.final_waypoints_start_ptr, self.
                                 lookahead_wps)
-        else:
-            if self.waypoints[self.final_waypoints_start_ptr].get_v() <\
-                    min(self.handoff_velocity, self.waypoints[self.final_waypoints_start_ptr].get_maxV()):
-                # Get the car going before handing off to JMT
-                if self.state != 'speedup':
-                    rospy.logwarn("Get Car moving from very slow speed at ptr={}.".format(self.final_waypoints_start_ptr))
-                offset =\
-                    self.init_acceleration(self.final_waypoints_start_ptr,
-                                           self.lookahead_wps)
-            if self.waypoints[self.final_waypoints_start_ptr].get_v() <\
-                    self.waypoints[self.final_waypoints_start_ptr].get_maxV():
-                recalc = self.generate_speedup(self.final_waypoints_start_ptr +
+            else:
+                if self.waypoints[self.final_waypoints_start_ptr].get_v() <\
+                        min(self.handoff_velocity, self.waypoints[self.final_waypoints_start_ptr].get_maxV()):
+                    # Get the car going before handing off to JMT
+                    if self.state != 'speedup':
+                        rospy.loginfo("Get Car moving from very slow speed at ptr={}.".format(self.final_waypoints_start_ptr))
+                    offset =\
+                        self.init_acceleration(self.final_waypoints_start_ptr,
+                                               self.lookahead_wps)
+                if self.waypoints[self.final_waypoints_start_ptr].get_v() <\
+                        self.waypoints[self.final_waypoints_start_ptr].get_maxV():
+                    recalc = self.generate_speedup(self.final_waypoints_start_ptr +
                                                offset,
                                                self.final_waypoints_start_ptr +
                                                self.lookahead_wps)
+        else:
+            rospy.logerr("Fell out of state engine - in state {} with dist_to_tl = {:3.2f}m and min_stoppig_distance {:3.2f}m"
+                         .format(self.state, dist_to_tl, self.min_stop_distance))
```

Comment:
May want to add `self.stopping_distance` to this log output for reference since it's used in the state thresholds.